packages/camera/camera_avfoundation/example/ios/RunnerTests/FLTCamSampleBufferTests.m

```diff
@@ -11,31 +11,98 @@
+  [_camera captureOutput:_camera.captureVideoOutput
+      didOutputSampleBuffer:_sampleBuffer
              fromConnection:OCMClassMock([AVCaptureConnection class])];
-  CVPixelBufferRef deliveriedPixelBuffer = [cam copyPixelBuffer];
+  CVPixelBufferRef deliveriedPixelBuffer = [_camera copyPixelBuffer];
   XCTAssertEqual(deliveriedPixelBuffer, capturedPixelBuffer,
                  @"FLTCam must deliver the latest captured pixel buffer to copyPixelBuffer API.");
-  CFRelease(capturedSampleBuffer);
   CFRelease(deliveriedPixelBuffer);
 }
 
+- (void)testFirstAppendedSampleShouldBeVideo {
+  id connectionMock = OCMClassMock([AVCaptureConnection class]);
+
+  id writerMock = OCMClassMock([AVAssetWriter class]);
+  OCMStub([writerMock alloc]).andReturn(writerMock);
+  OCMStub([writerMock initWithURL:OCMOCK_ANY fileType:OCMOCK_ANY error:[OCMArg setTo:nil]])
+      .andReturn(writerMock);
+  __block AVAssetWriterStatus status = AVAssetWriterStatusUnknown;
+  OCMStub([writerMock startWriting]).andDo(^(NSInvocation *invocation) {
+    status = AVAssetWriterStatusWriting;
+  });
+  OCMStub([writerMock status]).andDo(^(NSInvocation *invocation) {
+    [invocation setReturnValue:&status];
+  });
+
+  __block NSString *writtenSamples = @"";
+
+  id videoMock = OCMClassMock([AVAssetWriterInputPixelBufferAdaptor class]);
+  OCMStub([videoMock assetWriterInputPixelBufferAdaptorWithAssetWriterInput:OCMOCK_ANY
+                                                sourcePixelBufferAttributes:OCMOCK_ANY])
+      .andReturn(videoMock);
+  OCMStub([videoMock appendPixelBuffer:[OCMArg anyPointer] withPresentationTime:kCMTimeZero])
+      .ignoringNonObjectArgs()
+      .andDo(^(NSInvocation *invocation) {
+        writtenSamples = [writtenSamples stringByAppendingString:@"v"];
+      });
+
+  id audioMock = OCMClassMock([AVAssetWriterInput class]);
+  OCMStub([audioMock assetWriterInputWithMediaType:[OCMArg isEqual:AVMediaTypeAudio]
+                                    outputSettings:OCMOCK_ANY])
+      .andReturn(audioMock);
+  OCMStub([audioMock isReadyForMoreMediaData]).andReturn(YES);
+  OCMStub([audioMock appendSampleBuffer:[OCMArg anyPointer]]).andDo(^(NSInvocation *invocation) {
+    writtenSamples = [writtenSamples stringByAppendingString:@"a"];
+  });
+
+  FLTThreadSafeFlutterResult *result =
+      [[FLTThreadSafeFlutterResult alloc] initWithResult:^(id result){
+      }];
+  [_camera startVideoRecordingWithResult:result];
+
+  char *samples = "aaavava";
+
+  CMSampleBufferRef audioSampleBuffer = FLTCreateTestAudioSampleBuffer();
+  for (int i = 0; i < strlen(samples); i++) {
+    if (samples[i] == 'v') {
+      [_camera captureOutput:_camera.captureVideoOutput
+          didOutputSampleBuffer:_sampleBuffer
+                 fromConnection:connectionMock];
+    } else {
+      [_camera captureOutput:nil
+          didOutputSampleBuffer:audioSampleBuffer
+                 fromConnection:connectionMock];
+    }
+  }
```

Comment:
this test is unnecessarily complex. Can we just do a simpler version like this: 

didOutput: audio
didOutput: audio
didOutput: video

Then verify that the first 2 audio samples are omitted?

Reply:
I would rather do audio, audio, video, audio to check whether not just the first sample is omitted but also that not all audio samples are omitted at all. You also mean to avoid that for loop?

Reply:
yeah the for loop as well. keep the logic as simple as possible.